modules/sc-mesh-secure-deployment/src/nats/comms_nats_controller.py

```diff
@@ -308,26 +323,27 @@ def interface_monitor_cb(self, interfaces) -> None:
                            interface_name, operstate, mac_address.
         :return: -
         """
-        self.__comms_controller.logger.debug("Interface monitor cb: %s", interfaces)
-        self.__interfaces.clear()
-
-        for interface_data in interfaces:
-            interface = Interface(
-                interface_name=interface_data["interface_name"],
-                operstat=interface_data["operstate"],
-                mac_address=interface_data["mac_address"],
-            )
-            self.__interfaces.append(interface)
+        with self.__lock:
+            self.__comms_controller.logger.debug("Interface monitor cb: %s", interfaces)
+            self.__interfaces.clear()
+
+            for interface_data in interfaces:
+                interface = Interface(
+                    interface_name=interface_data["interface_name"],
+                    operstat=interface_data["operstate"],
+                    mac_address=interface_data["mac_address"],
+                )
+                self.__interfaces.append(interface)
 
-        if self.__if_to_monitor is None:
-            self.start_service_discovery()
-        else:
-            for interface in self.__interfaces:
-                if interface.interface_name == self.__if_to_monitor:
-                    if interface.operstat == "UP":
-                        self.start_service_discovery()
-                    elif interface.operstat == "DOWN":
-                        self.stop_service_discovery()
+            if self.__if_to_monitor is None:
```

Comment:
Can we already here check is the discovery running and not to call start if the discovery is running?
Just causing extra function call... minor thing.